frontend/src/lib/components/CommandBar/SearchResult.tsx

```diff
@@ -0,0 +1,86 @@
+import { useLayoutEffect, useRef } from 'react'
+import { useActions, useValues } from 'kea'
+
+import { resultTypeToName } from './constants'
+import { searchBarLogic, urlForResult } from './searchBarLogic'
+import { SearchResult as SearchResultType } from './types'
+import { LemonSkeleton } from '@posthog/lemon-ui'
+
+type SearchResultProps = {
+    result: SearchResultType
+    resultIndex: number
+    focused: boolean
+    keyboardFocused: boolean
+}
+
+const SearchResult = ({ result, resultIndex, focused, keyboardFocused }: SearchResultProps): JSX.Element => {
+    const { scrolling } = useValues(searchBarLogic)
+    const { onMouseEnterResult, onMouseLeaveResult, openResult, setScrolling } = useActions(searchBarLogic)
+
+    const ref = useRef<HTMLDivElement | null>(null)
+
+    useLayoutEffect(() => {
+        if (keyboardFocused) {
+            // :HACKY: This uses the non-standard scrollIntoViewIfNeeded api
+            // to improve scroll behaviour. Change to scrollIntoView({ scrollMode: 'if-needed' })
+            // once available.
+            if ((ref.current as any)?.scrollIntoViewIfNeeded) {
+                ;(ref.current as any).scrollIntoViewIfNeeded(false)
+            } else {
+                ref.current?.scrollIntoView()
+            }
+
+            // set scrolling state to prevent mouse enter/leave events during
+            // keyboard navigation
+            setScrolling(true)
```

Comment:
I'm not entirely sure what this does. I'm guessing it prevents mouse events when keyboard navigating? In that case maybe it should be renamed to `setTabbing` or `setNavigating`. I associate scrolling with a mouse based activity much more than a keyboard based one

Reply:
Yes exactly. When navigating via keyboard we scroll to the respective element (so that elements that are out of view can be navigated to). Maybe `isAutoScrolling`?

Reply:
It was mainly the word `scrolling` I was trying to avoid because I associate it the mouse being active not the keyboard. It's not a blocker though